support/testing/scenarios/power_test.go

```diff
@@ -0,0 +1,84 @@
+package scenarios_test
+
+import (
+	"context"
+	initactor "github.com/filecoin-project/specs-actors/actors/builtin/init"
+	"github.com/filecoin-project/specs-actors/actors/builtin/miner"
+	"github.com/filecoin-project/specs-actors/actors/util/adt"
+	"github.com/stretchr/testify/require"
+	"testing"
+
+	"github.com/filecoin-project/specs-actors/actors/abi"
+	"github.com/filecoin-project/specs-actors/actors/abi/big"
+	"github.com/filecoin-project/specs-actors/actors/builtin"
+	"github.com/filecoin-project/specs-actors/actors/builtin/power"
+	"github.com/filecoin-project/specs-actors/actors/runtime/exitcode"
+	"github.com/filecoin-project/specs-actors/support/testing/scenarios"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCreateMiner(t *testing.T) {
+	ctx := context.Background()
+	vm, addrs := scenarios.NewVMWithSingletons(ctx, t, 1)
+
+	params := power.CreateMinerParams{
+		Owner:         addrs[0],
+		Worker:        addrs[0],
+		SealProofType: abi.RegisteredSealProof_StackedDrg32GiBV1,
+		Peer:          abi.PeerID("not really a peer id"),
+	}
+	ret, code := vm.ApplyMessage(addrs[0], builtin.StoragePowerActorAddr, big.NewInt(1e10), builtin.MethodsPower.CreateMiner, &params)
+	assert.Equal(t, exitcode.Ok, code)
```

Comment:
We can do it in follow-up as we write more tests, but I suggest we make an API as ergonomic for test writing as we can. So, for example:
- add ApplyMessageOk that checks the exit code
- add GetState helper that checks errors already, and maybe returns values (that need casting) to save pre-allocation
- store methods that use a context the VM already has

Maybe we want a wrapper that encapsulates the `testing.T` for making all these checks (without pushing the T down into the VM proper)?

Reply:
I went ahead and implemented a higher level approach that tracks invocations in the VM and provides some infrastructure for making assertions against them. I think this flow through actors is what we should really focus on testing with scenarios.